lib/ansible/plugins/callback/slack_github.py

```diff
@@ -0,0 +1,153 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import json
+import os
+import subprocess
+from ansible.module_utils.urls import open_url
+from ansible.plugins.callback import CallbackBase
+
+
+class CallbackModule(CallbackBase):
+    """This is an ansible callback plugin that sends full
+    play recap to github gist and sends colored notification
+    to  a Slack channel after play ends.
+
+    This plugin makes use of file `slack.json` in your playbook folder
+    with following contents:
+    ```json
+    {
+        "webhook": "Slack incoming webhook url",
+        "username": "Ansible"
+    }
+    ```
+
+    Output: https://gist.github.com/anonymous/f75e04a115ea86b730e41e2ef03cc5f0
+    """
+
+    CALLBACK_VERSION = 2.0
+    CALLBACK_TYPE = 'notification'
+    CALLBACK_NAME = 'slack_github'
+    CALLBACK_NEEDS_WHITELIST = True
+
+    def __init__(self, display=None):
+        self.disabled = False
+        super(CallbackModule, self).__init__(display=display)
+        self.github_api = 'https://api.github.com/'
+        self._log = []
+
+    def init_config(self, playbook):
+        self.playbook_name = os.path.basename(playbook._file_name)
+        config_file = os.path.join(
+            playbook._basedir.encode(
+                'ascii', 'ignore'), 'slack.json')
```

Comment:
Instead of 'string'.encode() (or str('string') or bytes('string') ), use these ansible functions which are safer:
``` python
from ansible.module_utils._text import to_bytes

config_file = os.path.join(to_bytes(playbook._basedir, errors='surrogate_or_strict'), b'slack.json')
```

Some additional notes on that:
* to_bytes uses 'utf-8' by default which will handle more cases than 'ascii'.
* 'ignore' is not safe to use with filesystem paths.  A path is a sequence of bytes.  if you leave some of those bytes out then you end up with the name of a different file.  'surrogate_or_strict' will raise an exception if the bytes are not valid utf-8 which is unfortunate but better than using the wrong file.  
* I marked the second string, 'slack.json' explicitly as a byte string.  On  python3, string literals which are unmarked are text strings so you need to be explicit when writing most string literals.